radicle-cli/examples/rad-id-rebase.md

```diff
@@ -0,0 +1,430 @@
+
+ {
+   "payload": {
+     "xyz.radicle.project": {
+       "defaultBranch": "master",
+       "description": "Radicle Heartwood Protocol & Stack",
+       "name": "heartwood"
+     }
+   },
+   "delegates": [
+     "did:key:z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi",
+-    "did:key:z6MkedTZGJGqgQ2py2b8kGecfxdt2yRdHWF6JpaZC47fovFn"
++    "did:key:z6MkjchhfUsD6mmvni8mCdXHw216Xrm9bQe2mBH1P5RDjVJG"
+   ],
+   "threshold": 1
+ }
+
+
+--- Accepted ---
+
+total: 1
+keys: [
+  "z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi"
+]
+
+--- Rejected ---
+
+total: 0
+keys: []
+
+--- Quorum Reached ---
+
+✓ yes
+```
+
+Note that a warning was emitted:
+
+    ** Warning: Revision is out of date
+    ** Warning: d96f425412c9f8ad5d9a9a05c9831d0728e2338d =/= 475cdfbc8662853dd132ec564e4f5eb0f152dd7f
+    => Consider using 'rad id rebase' to update the proposal to the latest identity
+
+If we attempt to publish this revision, the command will fail:
+
+```
+$ rad id publish dad0a006af01b2ee61d3839b0d331b260729b545 --rev z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi/1 --no-confirm
+** Warning: Revision is out of date
+** Warning: d96f425412c9f8ad5d9a9a05c9831d0728e2338d =/= 475cdfbc8662853dd132ec564e4f5eb0f152dd7f
+=> Consider using 'rad id rebase' to update the proposal to the latest identity
+== Id failed
+the identity hashes do match 'd96f425412c9f8ad5d9a9a05c9831d0728e2338d =/= 475cdfbc8662853dd132ec564e4f5eb0f152dd7f' for the revision 'z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi/1'
+```
+
+So, let's fix this by running a rebase on the proposal's revision:
+
+```
+$ rad id rebase dad0a006af01b2ee61d3839b0d331b260729b545 --rev z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi/1 --no-confirm
+ok Identity proposal 'dad0a006af01b2ee61d3839b0d331b260729b545' rebased 🌱
+ok Revision 'z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi/4'
+title: Add Bob
+description: Add Bob as a delegate
+status:  open 
+author: z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi
+
+--- Document Diff ---
+
+ {
+   "payload": {
+     "xyz.radicle.project": {
+       "defaultBranch": "master",
+       "description": "Radicle Heartwood Protocol & Stack",
+       "name": "heartwood"
+     }
+   },
+   "delegates": [
+     "did:key:z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi",
+-    "did:key:z6MkedTZGJGqgQ2py2b8kGecfxdt2yRdHWF6JpaZC47fovFn"
++    "did:key:z6MkjchhfUsD6mmvni8mCdXHw216Xrm9bQe2mBH1P5RDjVJG"
+   ],
+   "threshold": 1
+ }
+
+
+--- Accepted ---
+
+total: 0
+keys: []
+
+--- Rejected ---
+
+total: 0
+keys: []
+
+--- Quorum Reached ---
+
+✗ no
+```
+
+We can now update the proposal to have both keys in the delegates set:
+
+```
+$ rad id update dad0a006af01b2ee61d3839b0d331b260729b545 --rev z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi/4 --delegates z6MkedTZGJGqgQ2py2b8kGecfxdt2yRdHWF6JpaZC47fovFn --no-confirm
+ok Identity proposal 'dad0a006af01b2ee61d3839b0d331b260729b545' updated 🌱
+ok Revision 'z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi/6'
```

Comment:
@cloudhead I have this oddity that I'm trying to figure out. When this `update` is run then we get back this revision as the last `OpId`, `z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi/6`.

If [this line below](https://github.com/radicle-dev/heartwood/pull/192/files#diff-c40c5d42e0913666973cf38374515886a882e2144cf5063b0e13d671f94037e1R345) is changed to use `z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi/6` then it says the revision was missing. I was able to confirm this by printing out the existing revisions which were:
* z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi/1
* z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi/4
* z6MknSLrJoTcukLrE435hVNQT4JUhbvWLX4kUzqkEStBU8Vi/5

So, what's happening to those other revisions? :thinking:

Reply:
Hmm keep in mind that creating a revision might use multiple actions, in which case those gaps in the clocks are expected.

Reply:
Hmmm ok, I'll have to check why the first revision is used as the one for the latest, i.e. why is `5` used instead of `6`

Reply:
I'm guessing the wrong op-id is returned by the revision creation process, will take a look.